app/models/core/MatchType.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.core
+
+import models.{Enumerable, WithName}
+
+sealed trait MatchType
```

Comment:
This is probably a cleaner design 

```scala
sealed trait MatchType {

  val isActiveTrader: Boolean = this match {
    case MatchType.FixedEstablishmentActiveNETP | MatchType.TraderIdActiveNETP | MatchType.OtherMSNETPActiveNETP => true
    case _ => false
  }

  val isQuarantinedTrader: Boolean = this match {
    case MatchType.FixedEstablishmentQuarantinedNETP | MatchType.TraderIdQuarantinedNETP | MatchType.OtherMSNETPQuarantinedNETP => true
    case _ => false
  }
}

```

This removes the need for CoreRegistrationValidationService to be involved in the decision making process as the value is a fixed calculation. Cuts down on meandering and mocking.